### PR TITLE
feat : manual coverage implemented binary search: closes #4

### LINKED ISSUE #4: feat : manual branch converage BinarySearch2dArray::BinarySearch
### What would you like to Propose?

complete manual branch coverage checks for the `BinarySearch` function

### Issue details

complete manual branch converage checks for the `BinarySearch` function

### Additional Information

coverage state before work: 88% Line coverage of the class

### PR DESCRIPTION
**Pull request by tjex**
In the BinarySearch2dArray class.
- added id branch tracking
- added print reached branches to console in Test file

I'm confused about one thing though, which is the sequence in which the ids are added to the HashSet. 
this is a print from a test run:
`branches reached: [8.1.1, 2.1, 8.1.2, 4.1, 5.1, 6.1, 7.1, 8.1, 7.2, 2.1.3, 2.1.2]`

`8.1` is a top level while loop. So I don't understand how `8.1.1` and `8.1.2` has been added to the HashSet before `8.1`...?

### CODE REVIEW

**Comment by einbergisak** [comment]
> `8.1` is a top level while loop. So I don't understand how `8.1.1` and `8.1.2` has been added to the HashSet before `8.1`...?

The HashSet is an unordered collection :)

File: src/main/java/com/thealgorithms/searches/BinarySearch2dArray.java [hunk 1]
```diff
@@ -73,13 +97,20 @@ static int[] binarySearch(
...
-            }; else if (arr[row][midIndex] < target) colStart =
-                midIndex + 1; else colEnd = midIndex - 1;
+                midIndex, }; 
+            } else if (arr[row][midIndex] < target) { 
+                branchesReached.add("8.1.2");
+                colStart = midIndex + 1; 
+            } else {
+                colEnd = midIndex - 1;
+            }
         }
```
Comment (einbergisak) [inline src/main/java/com/thealgorithms/searches/BinarySearch2dArray.java]: Coverage measurement should only concern the BinarySearch2dArray::BinarySearch method, and not BinarySearch2dArray::binarySearch (notice the initial capital B).

**Review by einbergisak — CHANGES_REQUESTED** [review]
Looks good, but should only have coverage for the appropriate method and not all methods of the class.

File: src/main/java/com/thealgorithms/searches/BinarySearch2dArray.java (hunk 1 above)
Comment (tjex) [reply src/main/java/com/thealgorithms/searches/BinarySearch2dArray.java]: @einbergisak fixed

**Review by einbergisak — CHANGES_REQUESTED** [review]
Looks good, but should have the "coverage" branch as target for merge, instead of "G17-development".

**Pull request merged**